com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/shader/ShaderCompilePipeline.java

```diff
@@ -78,14 +78,39 @@ private static int shaderLanguageToVersion(ShaderDesc.Language shaderLanguage) {
         };
     }
 
-    protected static boolean canBeCrossCompiled(ShaderDesc.Language shaderLanguage) {
+    protected static boolean shaderLanguageIsGLSL(ShaderDesc.Language shaderLanguage) {
         return shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM120 ||
                shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM140 ||
                shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM100 ||
                shaderLanguage == ShaderDesc.Language.LANGUAGE_GLES_SM300 ||
                shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM330 ||
-               shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM430 ||
-               shaderLanguage == ShaderDesc.Language.LANGUAGE_WGSL;
+               shaderLanguage == ShaderDesc.Language.LANGUAGE_GLSL_SM430;
+    }
+
+    protected static boolean canBeCrossCompiled(ShaderDesc.Language shaderLanguage) {
+        return shaderLanguageIsGLSL(shaderLanguage) || shaderLanguage == ShaderDesc.Language.LANGUAGE_WGSL;
+    }
+
+    private static byte[] remapTextureSamplers(ArrayList<SPIRVReflector.Resource> textures, String source) {
+        // Textures are remapped via spirv-cross according to:
+        //   SPIRV_Cross_Combined<TEXTURE_NAME><SAMPLER_NAME>
+        //
+        // Due to WGSL, before we pass the source into crosscompilation, we 'expand' samplers into texture + sampler:
+        //   uniform sampler2D my_texture;
+        // which becomes:
+        //   uniform texture2D my_texture;
+        //   uniform sampler   my_texture_separated;
+        //
+        // So these two rules together then becomes:
+        //   uniform sampler2D my_texture; ==> SPIRV_Cross_Combinedsampler_2dsampler_2d_separated
+        //
+        // Even without the separation of texture/sampler, we will still need to rename the texture in the source
+        // due to how spirv-cross works.
+        for (SPIRVReflector.Resource texture : textures) {
+            String spirvCrossSamplerName = String.format("SPIRV_Cross_Combined%s%s_separated", texture.name, texture.name);
+            source = source.replaceAll(spirvCrossSamplerName, texture.name);
+        }
```

Comment:
This will fix the problem where combined samplers (sampler2D, sampler3D, samplerArray, samplerCube) is renamed for the glsl shaders, but it doesn't support the case where a "sampler object" (i.e uniform sampler my_sampler) is expanded into a combined sampler2D. But we don't really support sampler objects in any capacity in the engine, so I've left it out for now.